packages/jest-mock/src/index.ts

```diff
@@ -1003,27 +1014,34 @@ export class ModuleMocker {
     return fn;
   }
 
-  spyOn<T extends {}, M extends NonFunctionPropertyNames<T>>(
+  spyOn<T extends object, M extends NonFunctionPropertyNames<T>>(
```

Comment:
`{}` as a constrain type allows any non-nullish value. See https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492. Here it is almost the same as `any`, because `string` or `boolean` are allowed. Might be better to use `object` (any non-primitive value).